clients/python/src/model_registry/_client.py

```diff
@@ -164,6 +166,73 @@ async def _register_model_artifact(
         return await self._api.upsert_model_version_artifact(
             ModelArtifact(name=name, uri=uri, **kwargs), mv.id
         )
+    
+    def _upload_to_s3(
+        self,
+        artifact_local_path: str,
+        destination: str,
+        region_name: str | None = None,
+    ) -> str:
+        """
+        Uploads a file to an S3 bucket.
+        """
+        pass
+
+    def _upload_to_oci(
+        self,
+        artifact_local_path: str,
+        destination: str,
+    ) -> str:
+        """
+        Uploads an artifact to an OCI registry.
+        """
+        pass
+    
+    def upload_artifact_and_register_model(
+        self,
+        name: str,
+        artifact_local_path: str,
+        destination_uri: str,
+        *,
+        version: str,
+        model_format_name: str,
+        model_format_version: str,
+        storage_key: str | None = None,
+        service_account_name: str | None = None,
+        author: str | None = None,
+        owner: str | None = None,
+        description: str | None = None,
+        metadata: Mapping[str, SupportedTypes] | None = None,
+        upload_client_params: Mapping[str, str] | None = None,
+    ) -> RegisteredModel:
+        if is_s3_uri(destination_uri):
+            self._upload_to_s3(artifact_local_path, destination_uri, upload_client_params['region_name'])
+        elif is_oci_uri(destination_uri):
+            self._upload_to_oci(artifact_local_path, destination_uri)
```

Comment:
I concur the flow is decided based on destination; just thinking we could leverage here functions to check the required configuration parameters (ie Env variables) are present

Reply:
I was thinking that the individual upload methods would be checking for those, as they would instantiate their respective clients (boto3 vs skopeo). 

I haven't done it yet in this PR, but the `upload_client_params` mapping will actually need to contain either the boto3 params OR the skopeo params, and that mapping is passed to both upload methods. Then those methods can determine whether it has enough information to instantiate a client and connect/perform the upload